e2e/test/scenarios/admin/datamodel/field.cy.spec.js

```diff
@@ -133,3 +135,52 @@ describe.skip("scenarios > admin > datamodel > field", () => {
     });
   });
 });
+    
+function getUnfoldJsonContent() {
+  return cy
+    .findByText("Unfold JSON")
+    .closest("section")
+    .find("[data-testid='select-button-content']")
+}
+
+describe("Unfold JSON", () => {
+  beforeEach(() => {
+    resetTestTable({ type: "postgres", table: "many_data_types" });
+    restore(`postgres-writable`);
+    cy.signInAsAdmin();
+    resyncDatabase({ dbId: WRITABLE_DB_ID, tableName: "many_data_types" });
+  });
+  
+  it("lets you enable/disable 'Unfold JSON' for JSON columns", () => {
+    cy.intercept("POST", `/api/database/${WRITABLE_DB_ID}/sync_schema`).as(
+      "sync_schema",
+    );
+    // Go to field settings
+    cy.visit(`/admin/datamodel/database/${WRITABLE_DB_ID}`);
+    cy.contains("Many Data Types").click();
```

Comment:
That should better be `cy.findByText("Many Data Types").click()`. If you need a substring, you can pass a regexp to `findByText`: `cy.findByText(/Many Data Types/i).click()`